Remove bootstrap modal uses and also remove some bootstrap CSS for buttons.

sahil839 opened the pull request:
<!-- Describe your pull request here.-->
- First commit is to update the modal on billing page.
- Second commit is to update modal on "/emails" page.
- Third commit is to remove bootstrap CSS for `.btn-primary`.
- Fourth commit is to remove bootstrap CSS for `.btn-success`.
- Fifth commit is to update the code to use `overlays` module instead of directly using `Micromodal`.
 <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
Before:
![emails-old](https://github.com/zulip/zulip/assets/35494118/528ce369-d313-404b-aedc-1568ac05f494)
![billing-old](https://github.com/zulip/zulip/assets/35494118/619cde86-d153-4810-bb0b-031b658aa66f)

After:
![Screenshot from 2023-10-04 21-15-16](https://github.com/zulip/zulip/assets/35494118/68b958f5-ae9a-4076-8de4-3530b84c9150)
![Screenshot from 2023-10-04 16-18-23](https://github.com/zulip/zulip/assets/35494118/8263869a-8cca-48cb-9b69-e3442c7536f7)



<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/src/billing/billing.ts:
```diff
@@ -1,5 +1,7 @@
 import $ from "jquery";
 
+import * as overlays from "../overlays";
```
I don't want to import `overlays.ts` in portico code; it has a lot of code that isn't designed to be used outside the app. I think a thing we could maybe plausibly do is split `overlays.ts` so that there's a separate `modals.ts` that is just the modals library -- that's probably an overdue split regardless.

But I think my preference would be to do something more isolated just using the upstream APIs, like we have in `web/src/portico/signup.ts`, if that doesn't require too much extra code, just because that isolation protects us from accidentally breaking the billing pages (which we aren't going to manually test often).

timabbott commented on web/src/portico/email_log.js:
```diff
@@ -46,4 +47,8 @@ $(() => {
+    $(".open-forward-email-modal").on("click", (e) => {
+        e.preventDefault();
+        overlays.open_modal("forward_email_modal");
+    });
```
This one I'm more OK with using `open_modal` in that it's not user-facing, so if some refactor breaks it, it's basically OK.

sahil839 commented:
Updated the PR to add `portico_modals.js` module which has `open_modal` and `close_modal` functions along with a couple of functions used by them. I have kept the checks that we use in `overlays` in these functions like closing a modal without opening it and others for now. Also, I have removed the code for `on_shown` and `on_hidden` callback as we do not use them currently in portico pages.

timabbott commented:
This is great, merged, thanks @sahil839!

Do we get to remove the Bootstrap modal library now? I guess if not let's discuss in #frontend, otherwise put up a PR.

sahil839 commented:
I think we can remove it, but want to confirm it once to avoid any bugs. Will open the PR or start a discussion if I have any doubts.
